src/Microsoft.AspNet.Mvc.TagHelpers/FormTagHelper.cs

```diff
@@ -124,6 +125,14 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             }
 
             var antiforgeryDefault = true;
+            if (output.Attributes.ContainsName("method"))
+            {
+                var methodValue = output.Attributes["method"].Value.ToString();
+                if (string.Equals(methodValue, FormMethod.Get.ToString(), StringComparison.OrdinalIgnoreCase))
+                {
+                    antiforgeryDefault = false;
+                }
+            }
```

Comment:
Whole block is miss-placed because `antiforgeryDefault` will be `false` anyhow if `output.Attributes.ContainsName(HtmlActionAttributeName)`. Move down below the `else` at [line 157](https://github.com/aspnet/Mvc/pull/3933/files#diff-8bd6f4dfc2f4a07e5b08261979376682R157).